[GOVCMS-5033] Support scaffold consolidation

stooit opened the pull request:
Provide a base for extensible SaaS codebases (to support scaffold consolidation effort).

Reduce/remove need for #137 until Satis or similar is available.

tobybellwood commented:
Great work, my 2c - govcms should be implementing drupal scaffolding inheritance from Drupal core, as we do in drupal-example-simple.  It's perfect for this use case, and extensible.  I'd be happy to give a demo.
Also, I tinkered with composer patchsets as a way of centrally managing patches for a known set of modules - it may be of use (https://github.com/tobybellwood/patchset)

stooit commented:
Agreed, unfortunately the change needs to be applied in the distro first (e.g your PR: https://github.com/govCMS/GovCMS8/pull/478) - we can't really move to `core-composer-scaffold` here as it conflicts with the legacy scaffold.

I have just simplified this PR a bit (and removed legacy scaffold, as it's really not necessary here) so it should make the pathway forward a little easier.

tobybellwood commented:
Yeah, that upstream PR is the unlocker. I've updated a few times for releases. Let me know if it needs any attention
